### DISCUSSION TITLE
log who killed you

### DISCUSSION THREAD

**Discussion by lonelybud** [YimMenu/YimMenu#2644] [Q&A]
i have tried hooking into native hooks or check existing network events.

I found this way to detect who killed you - 

```C++
static bool last_dead = false;

			if (g_local_player && g_local_player->m_player_info->m_game_state == eGameState::Died)
			{
				if (!last_dead)
				{
					auto ent = PED::GET_PED_SOURCE_OF_DEATH(self::ped);

					if (ent)
					{
						last_dead         = true;
						player_ptr player = nullptr;

						if (ENTITY::IS_ENTITY_A_PED(ent))
						{
							if (PED::IS_PED_A_PLAYER(ent))
								player = get_player_from_ped(ent);
						}
						else if (ENTITY::IS_ENTITY_A_VEHICLE(ent))
						{
							auto veh_ped = VEHICLE::GET_PED_IN_VEHICLE_SEAT(ent, -1, 0);

							if (PED::IS_PED_A_PLAYER(veh_ped))
								player = get_player_from_ped(veh_ped);
						}

						if (player)
							LOG(WARNING) << "You got Killed by: " << player->get_name();
					}
				}
			}
			else if (last_dead)
				last_dead = false;
```

it works like charm. Is there a better way?

**Comment by D-Borkan** [comment]
maybe for the best replace the Log and make him send notifaction
Edit : but the game already told you who kill you, but anyway this may good thing

**Comment by lonelybud** [comment]
actually this can be checked in CEventNetworkEntityDamage event in get_network_event_data hook.

`damage_data.m_victim_destroyed` should be true and victim should have player_info

**Reply by pierrelasse** [reply]
How 2 implement tho 😢

**Reply by lonelybud** [reply]
https://github.com/lonelybud/lonelyMenu/blob/924423bdc370880aedb8f08328a56d776d764362/src/hooks/info/get_network_event_data.cpp#L68-L82
